uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-calendar-day.component.js

```diff
@@ -64,46 +49,11 @@ define([
     }
 
     /**
-     * Returns the template name for Leave Requests tooltips.
-     *
-     * The pattern is `type-[days|hours]-on-[single-date|multiple-dates]-tooltip`
-     *
-     * @return {String}
-     */
-    function getTooltipTemplateForLeaveRequests () {
-      var dateRangeType, isSameDay;
-
-      isSameDay = moment(vm.contactData.leaveRequest.from_date)
-        .isSame(vm.contactData.leaveRequest.to_date, 'day');
-      dateRangeType = isSameDay ? 'single-date' : 'multiple-dates';
-
-      return 'type-' + vm.calculationUnit.name + '-on-' +
-        dateRangeType + '-tooltip';
-    }
-
-    /**
-     * Maps the absence type title to the leave request.
+     * Gets the absence type for the leave request.
      */
-    function mapLeaveRequestAbsenceType () {
-      absenceType = findRecordByIdFieldValue(vm.supportData.absenceTypes,
-        'id', vm.contactData.leaveRequest.type_id);
-
-      vm.contactData.leaveRequest['type_id.title'] = absenceType.title;
-    }
-
-    /**
-     * Maps the from and to date type labels to the leave request.
-     */
-    function mapLeaveRequestDateTypes () {
-      if (vm.calculationUnit.name === 'days') {
-        fromDateType = findRecordByIdFieldValue(vm.supportData.dayTypes,
-          'value', vm.contactData.leaveRequest.from_date_type);
-        toDateType = findRecordByIdFieldValue(vm.supportData.dayTypes,
-          'value', vm.contactData.leaveRequest.to_date_type);
-
-        vm.contactData.leaveRequest['from_date_type.label'] = fromDateType.label;
-        vm.contactData.leaveRequest['to_date_type.label'] = toDateType.label;
-      }
+    function getLeaveRequestAbsenceType (leaveRequest) {
+      return findRecordByIdFieldValue(vm.supportData.absenceTypes,
```

Comment:
I think you can use 
```js
return _.find(vm.supportData.absenceTypes, { id: leaveRequest.type_id });
```

There's one less function to remember that way.

Reply:
Nice!